src/macaron/dependency_analyzer/cyclonedx.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module contains helper functions to process CycloneDX SBOM."""
+
+import json
+import logging
+import os
+from collections.abc import Iterable
+from pathlib import Path
+from typing import Optional
+
+from macaron.config.global_config import global_config
+from macaron.dependency_analyzer.dependency_resolver import DependencyAnalyzer, DependencyInfo
+from macaron.errors import MacaronError
+from macaron.output_reporter.scm import SCMStatus
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class CycloneDXParserError(MacaronError):
+    """The CycloneDX error class."""
+
+
+def deserialize_bom_json(file_path: Path) -> dict:
+    """Deserialize the bom.json file.
+
+    Parameters
+    ----------
+    file_path : str
+        Path to the bom.json file.
+
+    Returns
+    -------
+    dict
+        The BOM content as a dictionary.
+
+    Raises
+    ------
+    CycloneDXParserError
+        If the bom.json file cannot be located or deserialized.
+
+    """
+    # TODO: use the official CycloneDX library to validate and deserialize BOM files
+    # once merged: https://github.com/CycloneDX/cyclonedx-python-lib/pull/290
+    if not os.path.exists(file_path):
+        raise CycloneDXParserError(f"Unable to locate the BOM file: {str(file_path)}.")
+
+    with open(file_path, encoding="utf8") as file:
+        try:
+            return dict(json.load(file))
+        except (ValueError, json.JSONDecodeError) as error:
+            raise CycloneDXParserError(f"Could not process the dependencies at {file_path}: {error}") from None
+
+
+def get_root_component(root_bom_path: Path) -> Optional[dict | None]:
+    """Get dependency components.
+
+    Parameters
+    ----------
+    root_bom_path : str
+        Path to the root bom.json file.
+
+    Returns
+    -------
+    Optional[dict|None]
+        The root component.
+    """
+    try:
+        root_bom = deserialize_bom_json(root_bom_path)
+        return root_bom.get("metadata").get("component")  # type: ignore
+    except (CycloneDXParserError, KeyError) as error:
+        logger.error(error)
+
+    return None
+
+
+def get_dep_components(
+    root_bom_path: Path, child_bom_paths: Optional[list[Path]] = None, recursive: bool = False
+) -> Iterable[dict]:
+    """Get dependency components.
+
+    Parameters
+    ----------
+    root_bom_path : str
+        Path to the root bom.json file.
+    child_bom_paths: Optional[list[Path]]
+        The list of paths to sub-project bom.json files.
+    recursive: bool
+        Set to False to get the direct dependencies only (default).
+
+    Yields
+    ------
+    dict
+        The dependencies as CycloneDX components.
+    """
+    bom_objects: list[dict] = []
+    try:
```

Comment:
Could we reuse the method `get_root_component` defined above to get the root component from `root_bom_path`?

Reply:
It can't be used directly because `get_root_component` returns `root_bom.get("metadata").get("component")` whereas here we need `root_bom.get("components")`.